test/cluster/test_view_building_coordinator.py

```diff
@@ -615,3 +617,121 @@ async def tablets_are_evenly_distributed():
+        await assert_row_count_on_host(cql, hosts[0], ks, "tab", 0)
+        await assert_row_count_on_host(cql, hosts[0], ks, "mv", 0)
+        await manager.server_start(servers[1].server_id)
+
+        # Repair the base table
+        s0_log = await manager.server_open_log(servers[0].server_id)
+        s0_mark = await s0_log.mark()
+        await manager.api.enable_injection(servers[0].ip_addr, "view_update_generator_consume_staging_sstable", one_shot=False)
+        await manager.api.repair(servers[0].ip_addr, ks, "tab")
+        await s0_log.wait_for(f"Processing {ks} failed for table tab", from_mark=s0_mark, timeout=60)
+        await s0_log.wait_for(f"Finished user-requested repair for tablet keyspace={ks}", from_mark=s0_mark, timeout=60)
+
+        await manager.server_stop_gracefully(servers[1].server_id)
+        await assert_row_count_on_host(cql, hosts[0], ks, "tab", 1000)
+        await assert_row_count_on_host(cql, hosts[0], ks, "mv", 0)
+        await manager.server_start(servers[1].server_id)
+
+        # Add node3
+        await manager.api.disable_tablet_balancing(servers[0].ip_addr)
+        new_server = await manager.server_add(cmdline=cmdline_loggers + ['--logger-log-level', 'view_update_generator=trace'], property_file={"dc": "dc1", "rack": "r1"})
+
+        s0_host_id = await manager.get_host_id(servers[0].server_id)
+        tablet_token = 0 # Doesn't matter since there is one tablet
+        async def get_tablet_replica_for_s0():
+            replicas = await get_tablet_replicas(manager, servers[0], ks, "tab", tablet_token)
+            for replica in replicas:
+                if replica[0] == s0_host_id:
+                    return replica
+            return None
+
+        s3_log = await manager.server_open_log(new_server.server_id)
+        s3_mark = await s3_log.mark()
+
+        # Move tablet from node0 to node3
+        src_replica = await get_tablet_replica_for_s0()
+        if not src_replica:
+            assert False, "no replica"
+        s3_host_id = await manager.get_host_id(new_server.server_id)
+        await manager.api.move_tablet(servers[0].ip_addr, ks, "tab", src_replica[0], src_replica[1], s3_host_id, 0, tablet_token)
+        # Without the fix for issue scylla-enterprise#4572 following line times out and ...
+        await s3_log.wait_for(f"Processed {ks}.tab:", from_mark=s3_mark, timeout=60)
+
+        new_hosts = await wait_for_cql_and_get_hosts(cql, [new_server], time.time() + 30)
+        await asyncio.gather(*(manager.server_stop_gracefully(s.server_id) for s in servers))
+        await assert_row_count_on_host(cql, new_hosts[0], ks, "tab", 1000)
+        # Start node0 here because view replica might not be migrated to the new node, since they are not colocated
+        await manager.server_start(servers[0].server_id)
+        # ... this check fails because the mv has 0 rows on the new host
+        await assert_row_count_on_host(cql, new_hosts[0], ks, "mv", 1000)
```

Comment:
Nit: the mv replica was not migrated to the new node, this query should have been sent to `servers[0]`. However, it's not a big deal because the new node will correctly route the request to `servers[0]` as the other tablet replica, `servers[1]` is down.

Consider sending a follow-up PR.